js-envs/src/main/scala/org/scalajs/jsenv/ExternalJSEnv.scala

```diff
@@ -25,9 +25,22 @@ abstract class ExternalJSEnv(
   /** Custom initialization scripts. */
   protected def customInitFiles(): Seq[VirtualJSFile] = Nil
 
-  protected class AbstractExtRunner(protected val classpath: CompleteClasspath,
-      protected val code: VirtualJSFile, protected val logger: Logger,
-      protected val console: JSConsole) {
+  protected class AbstractExtRunner(
+      protected val libs: Seq[ResolvedJSDependency],
+      protected val sjsCode: VirtualJSFile,
+      protected val code: VirtualJSFile) {
```

Comment:
> The NodeJSEnv for example makes a difference between sjsCode and code (how it loads it).

Does it really? All I see is `sjsCode` ends up in `getJSLibFiles()`, and `getJSLibFiles() :+ code` ends up in `getJSFiles()`, which itself is just a sequence of files fed to Node.js. Ultimately, they are treated 100% the same way.

PhantomJSEnv does seem to handle them differently.

> Conceptually, sjsCode is code that may not "do" anything. I.e. it may only define stuff.

Yes, we used to have such a distinction. I'm not sure it's still relevant, though.

Let's keep it that way for now, anyway.